src/middleware/user_auth_middleware.go

```diff
@@ -2,14 +2,45 @@ package middleware
 
 import (
 	"Bugs-Bunny/src/db"
+	"log"
+
 	"github.com/gofiber/fiber/v2"
 	uuid2 "github.com/google/uuid"
-	"log"
 )
 
 // UserAuthMiddleware is a middleware that checks if the user is authenticated
 func UserAuthMiddleware(c *fiber.Ctx) error {
 	// TODO: Implement User Auth Middleware
+	// Get the session token from the request headers
+	sessionToken := c.Get("Authorization")
+
+	// If the session token is not found, return an Unauthorized response
+	if sessionToken == "" {
+		return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+			"message": "Unauthorized. Missing authorization header.",
+		})
+	}
+	uuid, err := uuid2.Parse(sessionToken)
```

Comment:
token is always sent in Bearer Format:
```
Bearer JWT_TOKEN
```

This will always fail, we need to use strings.Split(sessionToken, " ")[1]

Reply:
Okay. I will rectify that.